lib/elixir/lib/kernel.ex

```diff
@@ -2176,17 +2176,17 @@ defmodule Kernel do
       true ->
         raise ArgumentError, "cannot set attribute @#{name} inside function/macro"
       false ->
-        arg = case name do
-          :behavior ->
+        cond do
+          name == :behavior ->
             :elixir_errors.warn env.line, env.file,
                                 "@behavior attribute is not supported, please use @behaviour instead"
-          :doc       -> {env.line, arg}
-          :typedoc   -> {env.line, arg}
-          :moduledoc -> {env.line, arg}
-          _          -> arg
+          :lists.member(name, [:moduledoc, :typedoc, :doc]) ->
+            {stack, _} = :elixir_quote.escape(env_stacktrace(env), false)
```

Comment:
I had to use the `:elixir_quote` module to escape the stack because `Macro.escape` is not compiled at this point during the bootstrap phase.